(but why? I didn't have this problem with the main branch)
```(candig) ~/Documents/GitHub/CANDIG/CanDIGv2 (fnguyen/clean-federation ✗) make build-federation
printf "\nOutput of build-federation: \n" >> tmp/error.txt
echo "    started build-federation" >> tmp/progress.txt
source setup_hosts.sh; \
        DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 \
        docker compose -f lib/candigv2/docker-compose.yml -f lib/federation/docker-compose.yml build  2>&1 | tee -a tmp/error.txt
IP Address is already set to: 192.168.4.69
#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 2B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 transferring context: 2B done
#2 DONE 0.0s
failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount2529679622/Dockerfile: no such file or directory
echo "    finished build-federation" >> tmp/progress.txt
```
```
(candig) ~/Documents/GitHub/CANDIG/CanDIGv2 (fnguyen/clean-federation ✗) make compose-federation
printf "\nOutput of compose-federation: \n" >> tmp/error.txt
echo "    started compose-federation" >> tmp/progress.txt
source lib/federation/federation_preflight.sh 2>&1 | tee -a tmp/error.txt
bash: lib/federation/federation_preflight.sh: No such file or directory
source setup_hosts.sh; \
        docker compose -f lib/candigv2/docker-compose.yml -f lib/federation/docker-compose.yml --compatibility up -d 2>&1 | tee -a tmp/error.txt
IP Address is already set to: 192.168.4.69
time="2023-05-31T11:28:55-07:00" level=warning msg="Found orphan containers ([candigv2_vault_1 candigv2_vault-runner_1 candigv2_opa-runner_1 candigv2_opa_1 candigv2_tyk_1 candigv2_tyk-redis_1 candigv2_keycloak_1 candigv2_candig-data-portal_1 candigv2_katsu_1 candigv2_metadata-db_1 candigv2_htsget_1 candigv2_minio_1]) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up."
 Container candigv2_federation_1  Created
 Container candigv2_federation_1  Starting
 Container candigv2_federation_1  Started
source lib/federation/federation_setup.sh 2>&1 | tee -a tmp/error.txt
>> waiting for federation to start
candigv2_federation_1
Adding servers to federation...
Traceback (most recent call last):
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/connection.py", line 200, in _new_conn
    sock = connection.create_connection(
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/util/connection.py", line 85, in create_connection
    raise err
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 61] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/connectionpool.py", line 790, in urlopen
    response = self._make_request(
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/connectionpool.py", line 496, in _make_request
    conn.request(
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/connection.py", line 388, in request
    self.endheaders()
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/http/client.py", line 1278, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/http/client.py", line 1038, in _send_output
    self.send(msg)
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/http/client.py", line 976, in send
    self.connect()
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/connection.py", line 236, in connect
    self.sock = self._new_conn()
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/connection.py", line 215, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x105c1c5e0>: Failed to establish a new connection: [Errno 61] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/connectionpool.py", line 844, in urlopen
    retries = retries.increment(
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/urllib3/util/retry.py", line 515, in increment
    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='candig.docker.internal', port=4232): Max retries exceeded with url: /federation/v1/servers (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x105c1c5e0>: Failed to establish a new connection: [Errno 61] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/lib/federation/initialize.py", line 65, in <module>
    main()
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/lib/federation/initialize.py", line 45, in main
    response = requests.request("POST", url, headers=headers, json=server)
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/Users/sonchau/Documents/GitHub/CANDIG/CanDIGv2/bin/miniconda3/envs/candig/lib/python3.10/site-packages/requests/adapters.py", line 519, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='candig.docker.internal', port=4232): Max retries exceeded with url: /federation/v1/servers (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x105c1c5e0>: Failed to establish a new connection: [Errno 61] Connection refused'))
echo "    finished compose-federation" >> tmp/progress.txt
```

--- a/Makefile.authx
+++ b/Makefile.authx
@@ -65,7 +65,6 @@ clean-tyk:
 #>>>
 # close federation service
 # make clean-federation
-
 #<<<
 clean-federation:
 	source setup_hosts.sh; \
@@ -90,6 +89,7 @@ clean-authx: clean-keycloak clean-tyk clean-vault clean-opa clean-federation
 #<<<
 init-authx: mkdir
 	$(MAKE) docker-volumes
+	$(foreach MODULE, $(CANDIG_AUTH_MODULES), $(MAKE) build-$(MODULE);)
 	$(foreach MODULE, $(CANDIG_AUTH_MODULES), $(MAKE) compose-$(MODULE);)
 
 

